pkg/controller/core/workload_controller.go

```diff
@@ -232,19 +295,25 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
 			log.V(2).Info("ClusterQueue for workload didn't exist; ignored for now")
 		}
 
-	case prevStatus == admitted && status == pending:
-		// trigger the move of associated inadmissibleWorkloads, if there are any.
-		r.queues.QueueAssociatedInadmissibleWorkloadsAfter(ctx, oldWl, func() {
-			// Delete the workload from cache while holding the queues lock
-			// to guarantee that requeueued workloads are taken into account before
-			// the next scheduling cycle.
-			if err := r.cache.DeleteWorkload(oldWl); err != nil {
-				log.Error(err, "Failed to delete workload from cache")
+	case prevStatus == cancellingAdmission && status == pending:
+		cqName, cqOk := r.queues.ClusterQueueForWorkload(wl)
```

Comment:
it makes more sense to calculate this inside `DeleteWorkload` and `QueueAssociatedInadmissibleWorkloadsAfter`. The cache also has information about queues.

Maybe revert the changes in this file (other than adding the action) https://github.com/kubernetes-sigs/kueue/pull/512/files#diff-22ad2263a86a607fd28df7741c704614d0f34e208b5270153aa39427e4325fb3L351

Reply:
Done